pkg/controllers/sync/controller.go

```diff
@@ -409,7 +409,11 @@ func (s *SyncController) reconcile(ctx context.Context, federatedName common.Qua
 		keyedLogger.Error(err, "Failed to get pending controllers")
 		return worker.StatusError
 	}
-	if len(pendingControllers) > 0 {
+	isFedHPAObject, err := isFedHPAObject(ctx, fedResource.Object())
+	if err != nil {
+		return worker.StatusError
+	}
+	if len(pendingControllers) > 0 && !isFedHPAObject {
```

Comment:
Why do we leave a backdoor for `isFedHPAObject` to pass this check and then set its placement to nil? Is it sufficient to simply block its propagation here?

Reply:
The function **isFedHPAObject** naming here is somewhat ambiguous. I'll make some modifications.    The intention is to clean up the HPAs previously deployed in member clusters when the centralized HPA takes effect, in order to avoid conflicts between the centralized HPA in the federation and the HPAs in the member clusters.   Do you have any better ideas.

Reply:
I have tested it, and it works correctly. When the HPA is working in centralized mode, it will clean up the HPAs that have been distributed to member clusters.